src/Layouts/body/Body.jsx

```diff
@@ -1,20 +1,73 @@
-import React from "react"
-import GenreFilter from "../../Components/UI/genreFilter/GenreFilter"
-import ResultSort from "../../Components/UI/resultSort/ResultSort"
-import VideosList from "../../Components/VideosList"
+import React, { useState } from "react";
+import GenreFilter from "../../Components/UI/genreFilter/GenreFilter";
+import ResultSort from "../../Components/UI/resultSort/ResultSort";
+import VideosList from "../../Components/VideosList";
+import VidoeModal from '../../Components/UI/videomodal/VideoModal'
+import VideoForm from '../../Components/UI/videoform/VideoForm'
+import MyButton from '../../Components/UI/button/MyButton'
 
-const Body = () => {
+const Body = ({ videos, setVideos }) => {
+
+  const [videoObject, setVideoObject] = useState();
+  const [editVieoModal, setEditVieoModal] = useState(false);
+  const [deleteModal, setDeleteModal] = useState(false);
+  const [selectedSort, setSelectedSort] = useState('');
+
+  const sortVideos = (sort) => {
+    setSelectedSort(sort);
+    setVideos([...videos].sort((a, b) => a[sort].localeCompare(b[sort])));
+  };
+
+  const showEditVideoModal = (video, show) => {
+    setVideoObject(video);
+    setEditVieoModal(show);
+  };
+
+  const showDeleteModal = (video, show) => {
+    setVideoObject(video);
+    setDeleteModal(show);
+  };
+
+  const editVideo = (video) => {
+    const editedVideos = videos.map(vd => {
+      if (vd.id == video.id) {
+        return video;
+      }
+
+      return vd;
+    });
+
+    setVideos(editedVideos);
+  }
+
+  const removeVideo = (video) => {
+    setVideos(videos.filter(v => v.id !== video.id));
+    setDeleteModal(false);
+  };
+  
   return(
-    <main className="main">
-      <section className="container">
-        <div className="sort">
-          <GenreFilter />
-          <ResultSort />
-        </div>
-      </section>
-      <VideosList />
-    </main>
+    <>
+      <VidoeModal className='videoModal' title='edit movie' visible={editVieoModal} setVisible={setEditVieoModal}>
+        <VideoForm editVideo={videoObject} creat={editVideo} showModal={setEditVieoModal} />
+      </VidoeModal>
+      <VidoeModal className='videoModal' title='delete movie' visible={deleteModal} setVisible={setDeleteModal}>
+        <p>Are you sure you want to delete this movie?</p>
```

Comment:
I know that we have this description only in one modal window from design, but if this was a real project, it would be highly possible that we would use it for other modals in future... so I would suggest to make it as a part of Modal component as well, as an optional property